### PR TITLE
CU-1tmym8b - Refactor crowdloan rewards

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
- unsigned origin for association
- check for already associated account
- remove legacy <Bytes> prefix
- refactor tests/benchmarks to reuse structures/code

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-1tmym8b Refactor crowdloan reward pallet for release](https://app.clickup.com/t/1tmym8b)

**Review by cor — DISMISSED** [review]
lgtm (pair programmed)

File: frame/crowdloan-rewards/src/tests.rs [hunk 1]
```diff
@@ -436,37 +321,14 @@ mod test_prevalidate_association {
...
-				assert!(matches!(
-					call.dispatch(Origin::root()),
-					Ok(frame_support::dispatch::PostDispatchInfo {
-						actual_weight: _,
-						pays_fee: Pays::No
-					})
-				));
-			}
-		});
-	}
```
Comment (hussein-aitlahcen) [inline frame/crowdloan-rewards/src/tests.rs:460]: Not required as it is an unsigned extrinsic now.

File: frame/crowdloan-rewards/src/lib.rs [hunk 2]
```diff
@@ -238,14 +240,14 @@ pub mod pallet {
...
-		#[pallet::weight((<T as Config>::WeightInfo::associate(TotalContributors::<T>::get()), Pays::No))]
+		#[pallet::weight(<T as Config>::WeightInfo::associate(TotalContributors::<T>::get()))]
 		#[transactional]
 		pub fn associate(
 			origin: OriginFor<T>,
 			reward_account: T::AccountId,
 			proof: ProofOf<T>,
 		) -> DispatchResultWithPostInfo {
-			T::AssociationOrigin::ensure_origin(origin)?;
+			ensure_none(origin)?;
```
Comment (KaiserKarel) [inline frame/crowdloan-rewards/src/lib.rs:250]: Are regular users able to provide Origin::None? Shouldn't this just be ensure_signed, but not using the account_id provided?

Comment (hussein-aitlahcen) [reply frame/crowdloan-rewards/src/lib.rs:250]: `ensure_none` just ensure that the extrinsic has not been signed from what the doc is describing. I've seen that you can send unsigned extrinsic using polkadotjs

Comment (hussein-aitlahcen) [reply frame/crowdloan-rewards/src/lib.rs:250]: [It looks like it is the intended behavior](https://github.com/paritytech/polkadot/blob/master/runtime/common/src/claims.rs#L317)

**Review by KaiserKarel — APPROVED** [review]

**Review by andor0 — APPROVED** [review]

**Pull request merged**